### PR TITLE
Simplifies creating test API clients

### PR DESCRIPTION
**Pull request by AngellusMortis**
Refactors @bdraco mentioned.

I am not sure why it is complaining about assigning to "Cannot assign to a method" now for the replacing the methods. I also cannot find a proper error code for that either to disable it via config since it is pretty common in tests.

### CODE REVIEW

**Comment by bdraco** [comment]
Where are you seeing the Cannot assign to a method?

File: tests/conftest.py [hunk 1]
```diff
@@ -139,54 +142,43 @@ def ensure_debug():
...
-@pytest.mark.asyncio
-async def old_protect_client():
-    client = UpvServer(None, "127.0.0.1", 0, "username", "password")
-    client.api_request = AsyncMock(side_effect=mock_api_request)
-    client.api_request_raw = AsyncMock(side_effect=mock_api_request_raw)
-    client.ensure_authenticated = AsyncMock()
+async def setup_client(client: Union[ProtectApiClient, UpvServer], websocket: SimpleMockWebsocket):
+    client.api_request = AsyncMock(side_effect=mock_api_request)  # type: ignore
+    client.api_request_raw = AsyncMock(side_effect=mock_api_request_raw)  # type: ignore
+    client.ensure_authenticated = AsyncMock()  # type: ignore
```
Comment (AngellusMortis) [inline tests/conftest.py:148]: Where I added the `type: ignore` comments in `setup_client`. I get them when I run mypy locally.

**Comment by bdraco** [comment]
Thanks.  Will take a look.  May be tomorrow since I’m pretty wiped from travel.

File: tests/conftest.py (hunk 1 above)
Comment (bdraco) [reply tests/conftest.py:148]: Looks like its https://github.com/python/mypy/issues/6713

**Review by bdraco — APPROVED** [review]

**Pull request merged**